pkg/provision/mbus/api.go

```diff
@@ -85,6 +88,42 @@ func (n *Deployments) listPublicIps(ctx context.Context, _ []byte) (interface{},
 	return ips, nil
 }
 
+type listPrivateIpsArgs struct {
+	NetworkName string `json:"network_name"`
+}
+
+func (d *Deployments) listPrivateIps(ctx context.Context, payload []byte) (interface{}, error) {
+	var args listPrivateIpsArgs
+	if err := json.Unmarshal(payload, &args); err != nil {
+		return nil, err
+	}
+	deployments, err := d.list(ctx, payload)
+	if err != nil {
+		return nil, err.Err()
+	}
+	dls := deployments.([]gridtypes.Deployment)
```

Comment:
Why assertion? it's not a good idea to do type assertion because changes to the list return type can break this in runtime and the compiler will not know about it. Instead do the following:
```
fn listDeployments(ctx, twin_id) []Depoyments 
fn list() {
    return listDeployments(ctx, GetTwinId(ctx))
}
fn list_piv_ips() {
    deployments = listDeployments(ctx, GetTwinId(ctx))
    // work with deployments
}
```